### PR TITLE
add provider attribute for specifying sonarqube version

### PR DESCRIPTION
**Pull request by mrparkers**
I was running into an issue when using Terraform to install a Sonarqube server (via the `helm_release` resource), then using this provider to configure it.  The problem with this is that the provider initialization process fetches the version from the Sonarqube server, and this fails during the first run of `terraform apply` because the Sonarqube server is not running yet.

I added this provider attribute so users could specify the version up front, which avoids this initial request.  This allows my Terraform code to both install the Sonarqube server, and use this provider to configure it (all with one run of `terraform apply`).

There is probably a better way to do this, but this works for now.  I also backfilled some missing docs from my last PR.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -17,3 +17,4 @@ The following arguments are supported:
 - user - (Required) Sonarqube user. This can also be set via the SONARQUBE_USER environment variable.
 - pass - (Required) Sonarqube pass. This can also be set via the SONARQUBE_PASS environment variable.
 - host - (Required) Sonarqube url. This can be also be set via the SONARQUBE_HOST environment variable.
+- installed_version - (Optional) The version of the Sonarqube server. When specified, the provider will avoid requesting this from the server during the initialization process. This can be helpful when using the same Terraform code to install Sonarqube and configure it.
```
Comment (mrparkers) [inline docs/index.md:20]: I wanted this to be `version`, but that's a reserved provider attribute.

**Comment by jdamata** [comment]
I'm trying to understand the issue. 

You are installing sonarqube into kubernetes via the kubernetes terraform provider and also configuring it at the same time. The issue arises on the first terraform run due to the sonarqube installation not being ready when this provider attempts to create some configuration.

Wouldn't this still be a problem with the changes in this pr but instead of the initialization failing, the resources would fail to create because the sonarqube server is not ready?

**Comment by jdamata** [comment]
Does the terraform installation of sonarqube poll for the readiness of sonarqube before attempting to apply the configuration? If so i can see how this PR helps.

Either way, I think I am ok with merging this.

**Comment by mrparkers** [comment]
> You are installing sonarqube into kubernetes via the kubernetes terraform provider and also configuring it at the same time. The issue arises on the first terraform run due to the sonarqube installation not being ready when this provider attempts to create some configuration.

Yup, you got it.

> Wouldn't this still be a problem with the changes in this pr but instead of the initialization failing, the resources would fail to create because the sonarqube server is not ready?

Not quite.  It really depends on the Terraform code you're trying to run.  In my case, I have a module that installs Sonarqube to a Kubernetes cluster, and another module that uses this provider to configure it.  The second module uses `depends_on` to ensure that the first module is finished before moving on to the second one.

The other thing worth mentioning is that the provider initialization step happens several times throughout the course of a single `terraform apply`.  I don't know the exact logic or path it follows, but you can see this when running your acceptance tests by looking at the number of times the version lookup happens during a single test.  At the very least, provider initialization happens during the implicit plan from `terraform apply`, and again before the resources need to be created.  So by the time the resources need to be created, the provider is populated with the information needed to authenticate to Sonarqube.

> Does the terraform installation of sonarqube poll for the readiness of sonarqube before attempting to apply the configuration? If so i can see how this PR helps.

Yeah, the `helm_release` resource waits until all of the replicas for the Sonarqube application are ready before finishing.  By using `depends_on` for my configuration module, I can wait until it's ready before attempting to configure it using this provider.

**Comment by mrparkers** [comment]
Another approach that may also work here is adding a provider attribute that specifies whether or not the health check should occur during the provider initialization phase.  The provider I work on has an [`initial_login`](https://registry.terraform.io/providers/mrparkers/keycloak/latest/docs#initial_login) attribute that essentially does this, for the exact same use case as I'm describing here.

I initially tried this approach, but it ended up becoming a larger refactor than I was comfortable with, due to the need to check which server version we're operating against for a lot of resources.  If I had more time, I'd probably attempt this again by separating the code that's responsible for communicating with Sonarqube into its own package, so the version checking logic could be encapsulated in one place.  I might be able to give this a shot some time next week if you're interested.

**Comment by jdamata** [comment]
> f I had more time, I'd probably attempt this again by separating the code that's responsible for communicating with Sonarqube into its own package, so the version checking logic could be encapsulated in one place. I might be able to give this a shot some time next week if you're interested.

Sounds like something i should have done in the initial implementation. If you've got the time/motivation to make a change like this i'd be happy to review/approve it!

No idea why drone isn't kicking off a build but the build should kick off once this is merged anyways. 

Thanks for the docs and this PR!!! Appreciate it dude ^_^

**Pull request merged**